src/Entity/AbstractToken.php

```diff
@@ -0,0 +1,224 @@
+<?php
+/*
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * This software consists of voluntary contributions made by many individuals
+ * and is licensed under the MIT license.
+ */
+
+namespace ZfrOAuth2\Server\Doctrine\Entity;
+
+use DateTime;
+
+/**
+ * Provide basic functionality for both access tokens, refresh tokens and authorization codes
+ *
+ * Please note that scopes are stored as a saved as a string instead using associations to scope entities, mainly
+ * for performance reasons and to avoid useless database calls
+ *
+ * @author  Michaël Gallego <mic.gallego@gmail.com>
+ * @licence MIT
+ */
+abstract class AbstractToken
```

Comment:
Ive got or working with using the model as entities. (Getting a access token with default scope for a password grant)

But only if i set AbstractToken::expiresAt as protected. Other properties can be private. I don't know why that is. I'll push a bit later maybe ocramius know what's going on there...





> Op 31 jan. 2016 om 13:47 heeft Michaël Gallego <notifications@github.com> het volgende geschreven:
> 
> In src/Entity/AbstractToken.php:
> 
> > + */
> > +
> > +namespace ZfrOAuth2\Server\Doctrine\Entity;
> > +
> > +use DateTime;
> > +
> > +/**
> > + * Provide basic functionality for both access tokens, refresh tokens and authorization codes
> > + *
> > + * Please note that scopes are stored as a saved as a string instead using associations to scope entities, mainly
> > + * for performance reasons and to avoid useless database calls
> > + *
> > + * @author  Michaël Gallego <mic.gallego@gmail.com>
> > + * @licence MIT
> > + */
> > +abstract class AbstractToken
> Doctrine bypass this by using Reflection. And remember that we have the "reconstitue" method, so the idea is that the repository fetch data from database, and call the "Token::reconstitue($data)", so instances can be created.
> 
> —
> Reply to this email directly or view it on GitHub.
>